backend/estimate-service/src/main/java/org/egov/works/repository/IdGenRepository.java

```diff
@@ -0,0 +1,67 @@
+package org.egov.works.repository;
+
+
+import digit.models.coremodels.IdGenerationRequest;
+import digit.models.coremodels.IdGenerationResponse;
+import digit.models.coremodels.IdRequest;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.tracer.model.CustomException;
+import org.egov.tracer.model.ServiceCallException;
+import org.egov.works.config.EstimateServiceConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+public class IdGenRepository {
+
+
+    private RestTemplate restTemplate;
+
+    private EstimateServiceConfiguration config;
+
+    @Autowired
+    public IdGenRepository(RestTemplate restTemplate, EstimateServiceConfiguration config) {
```

Comment:
Use @Autowired to initialize instead of using constructor, wherever possible.

Reply:
done